language/diem-vm/src/diem_transaction_executor.rs

```diff
@@ -753,7 +764,7 @@ fn is_reconfiguration(vm_output: &TransactionOutput) -> bool {
 /// Waypoints and BlockPrologues are not signed and are unaffected by signature checking,
 /// but a user transaction or writeset transaction is transformed to a SignatureCheckedTransaction.
 #[derive(Debug)]
-enum PreprocessedTransaction {
+pub(crate) enum PreprocessedTransaction {
```

Comment:
What's this change for? I'm guessing so you can experiment with calling execute_single_transaction from elsewhere in the crate, and I guess that's OK temporarily. I'm not a big fan of this enum, though -- it's unnecessarily confusing the way that it renames BlockMetadata to BlockPrologue and GenesisTransaction to WaypointWriteSet. I'd like to see us get rid of this instead, but as long as we keep it within the crate, I don't object.

Reply:
That is exactly right. My intention for this PR is to separate out a clean api for executing one single transaction so that I can implement my parallel execution refactor on top.